allennlp/data/iterators/same_language_iterator.py

```diff
@@ -22,7 +25,7 @@ def split_by_language(instance_list):
 
 
 @DataIterator.register("same_language")
-class SameLanguageIterator(BucketIterator):
+class SameLanguageIteratorStub:
```

Comment:
For backward compatibility, yeah I don't know why both these iterators exist in the first place, I guess we just missed the similarity in #2628.

Thanks, I missed that stub vs shim thing.

Reply:
We have two of them because they were landed at about the same time and we did not manage to synchronize.

While ``homogenous batch`` is more general, ``same lang`` supports bucketing. ``homogenous batch`` with bucketing would be the best of two worlds.